test/tools/ossfuzz/protoToAbiV2.cpp

```diff
@@ -370,26 +466,33 @@ string ProtoConverter::testCode(unsigned _invalidLength)
 		</atLeastOneVar>
 		return 0;
 		)")
-		("parameterNames", dev::suffixedVariableNameList(s_varNamePrefix, 0, m_varCounter))
+		("argumentNames", m_argsCoder.str())
 		("invalidLengthFuzz", std::to_string(_invalidLength))
 		("isRightPadded", isLastDynParamRightPadded() ? "true" : "false")
 		("atLeastOneVar", m_varCounter > 0)
 		.render();
 }
 
-string ProtoConverter::helperFunctions()
+string ProtoConverter::testReturnDataFunction()
 {
-	stringstream helperFuncs;
-	helperFuncs << R"(
-	function bytesCompare(bytes memory a, bytes memory b) internal pure returns (bool) {
-		if(a.length != b.length)
-			return false;
-		for (uint i = 0; i < a.length; i++)
-			if (a[i] != b[i])
-				return false;
-		return true;
-	}
+	return Whiskers(R"(<?varsPresent>
+		(<varDecl>) = this.coder_returndata_public();
+<equality_checks>
+		(<varRef>) = this.coder_returndata_external();
+<equality_checks></varsPresent>
+		return 0;
+		)")
+		("varsPresent", !m_typedReturn.str().empty())
+		("varDecl", m_typedReturn.str())
+		("equality_checks", regex_replace(m_checks.str(), regex("p_"), "lv_"))
```

Comment:
Is there any reason not to prefix those variables with ``p_`` as well when generating ``m_typedReturn`` above and that way get rid of the ``regex_replace`` here? If it's not really necessary it seems a bit like overkill here...

Reply:
Ah, you are right. I think the reason I used regex was because in an earlier version of this PR, I was generating both calldata and returndata test functions in a single test case. Now that I generate either-or, regex can be removed.